pkg/storage/storage.go

```diff
@@ -107,6 +107,14 @@ func (s *Storage) ListDeployed() ([]*rspb.Release, error) {
 	})
 }
 
+// NoDeployedReleases describes the situation where none of the releases in the history are deployed.
+//
+// Aggressive pruning can mask the fact that there is a deployed release, but that it
+// was cleaned up by the history pruner.
+type NoDeployedReleases string
+
+func (e NoDeployedReleases) Error() string { return string(e) }
```

Comment:
Assuming we're ok with targeting go 1.13, we could use wrapped errors: https://blog.golang.org/go1.13-errors

https://play.golang.org/p/4ryS6CCiPCa